File: API/Helpers/Params/UserParams.cs
namespace API.Helpers {
    public class UserParams : PaginationParams {
        public string CurrentUsername { get; set; }
        public string Gender { get; set; } = "any";
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 100;
        public string OrderBy { get; set; } = "lastActive";
        public bool PhotoRequired { get; set; } = false;
    }
}

File: client-app/src/app/helpers/models/userParams.ts
import { User } from "./user";

export class UserParams {
    gender = "any";
    minAge = 18;
    maxAge = 99;
    pageNumber = 1;
    pageSize = 12;
    orderBy = 'lastActive';
    photoRequired = true;

    constructor(user: User) {
        // this.gender = user.gender === 'female' ? 'male' : 'female';
        // this.gender = 'any';
    }
}